public/components/agents/fim/inventory/fileDetail.tsx

```diff
@@ -111,12 +108,51 @@ export class FileDetails extends Component {
         name: 'Permissions',
         icon: 'lock',
         link: false,
+        transformValue: (value, {agent}) => {
```

Comment:
You could refactor this part of the code

Create a function to pass to the "transformValues" property that looks like this:

```ts
transformValue: (value, {agent}) => newMethod(value, {agent})

newMethod(value, { agent }) {
  if (((agent || {}).os || {}).platform === 'windows' && value && value !== '-') {
    const components = value.split(', ')
      .map(createPermisionObject)
      .sort(sortPermisionObject)
      .map(createPermisionComponent)
    return (<TruncateHorizontalComponents
      components={components}
      labelButtonHideComponents={(count) => `+${count} users`}
      buttonProps={{ size: 'xs' }}
      componentsWidthPercentage={0.85}
    />)
  };
  return value;
}
```

Reply:
I moved the logic to class methods to render the file details of `Permissions` and `Size`.